daffodil-lib/src/main/scala/org/apache/daffodil/lib/xml/XMLUtils.scala

```diff
@@ -1113,6 +1189,29 @@ Differences were (path, expected, actual):
         val b = DFDLDateTimeConversion.fromXMLString(dataB)
         a == b
       }
+      case Some("xs:double") => {
+        val a = strToDouble(dataA)
+        val b = strToDouble(dataB)
```

Comment:
Mentioned above, but this could be `NodeInfo.Double.fromXMLString(...)`. Same with xs:float below.

Reply:
Same resolution. NodeInfo.Double.fromString calls this.